python-test/features/agentsProvider.feature

```diff
@@ -7,3 +7,17 @@ Feature: agent provider
             And the agent container is started
         Then the agent status in Orb should be online
             And the container logs should contain the message "sending capabilities" within 10 seconds
+
+
+    Scenario: Apply two policies to an agent
+        Given the Orb user logs in
+            And that an agent already exists and is online
+            And referred agent is subscribed to a group
+            And that a sink already exists
+            And that a policy already exists
+            And that a dataset using referred group, sink and policy ID already exists
+            And that container logs contains messages "scraped metrics for policy" and "managing agent policy from core" for applied policy within 180 seconds
+        When a new policy is created
+            And a new dataset is created using referred group, sink and policy ID
+        Then the container logs should contain messages "scraped metrics for policy" and "managing agent policy from core" for applied policy within 180 seconds
```

Comment:
I guees you should check if the new policy applied it's scrapping metrics... you can check the policy id / name on the logs... It should be present the two policies ids / names (e.g. 2021-12-27T11:20:20.118-0300	INFO	pktvisor/pktvisor.go:323	scraped metrics for policy	{"policy": "policy", "policy_id": "939a3af0-50d5-4338-ad43-2f1e9e68f7ca", "payload_size_b": 1829}
2021-12-27T11:20:20.118-0300	INFO	pktvisor/pktvisor.go:323	scraped metrics for policy	{"policy": "policy2", "policy_id": "10c3f357-a3bb-41f5-a829-d8358345df06", "payload_size_b": 1838})

Reply:
why 2 different policy IDs?
I see that `check_logs_contain_message_for_policy()` checks the logs like you mentioned.

Reply:
dscabral is correct - the goal of this test is to make sure that a second policy is applied and scraped correctly. unless we check for each policy separately, this test is not functioning correctly. but, it does appear that it tries to check id per policy.

Reply:
tests now verify that all policies applied to the agent are actually applied and that they all publish metrics after the last policy was applied